### PR TITLE
Porting fileservers from salt

### PR DESCRIPTION
**Pull request by devagupt**
- Porting various fileservers (AzureFS, GitFS, RootFS, S3FS) and their dependencies in salt
- Ported test cases of all applicable modules/utils.

### CODE REVIEW

File: hubblestack/config/__init__.py [hunk 1]
```diff
@@ -0,0 +1,1877 @@
...
+import sys
+import glob
+import time
+import codecs
+import logging
+import types
+from copy import deepcopy
+
+# pylint: disable=import-error,no-name-in-module
+from salt.ext import six
```
Comment (goravsingal) [inline hubblestack/config/__init__.py]: No Six module.
All salt.xyz dependencies must be removed below.

And looks like we have copied code which we don't need. Example zeromq.

File: hubblestack/payload.py [hunk 2]
```diff
@@ -0,0 +1,244 @@
+# -*- coding: utf-8 -*-
+"""
+Many aspects of the salt payload need to be managed, from the return of
+encrypted keys to general payload dynamics and packaging, these happen
+in here
+"""
+
+# Import python libs
+from __future__ import absolute_import, print_function, unicode_literals
```
Comment (goravsingal) [inline hubblestack/payload.py]: I think this module is assigned to someone else in wiki.
Either you can remove this, or mail to him/her telling that you have done this.

File: hubblestack/payload.py [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+
+import hubblestack.log
+import hubblestack.utils.stringutils
+import hubblestack.transport.frame
+import hubblestack.utils.immutabletypes as immutabletypes
+from hubblestack.utils.exceptions import HubbleDeserializationError
+from hubblestack.utils.data import CaseInsensitiveDict
+
+try:
+    import zmq
```
Comment (goravsingal) [inline hubblestack/payload.py]: We dont need zero mq

File: hubblestack/transport/ipc.py [hunk 4]
```diff
@@ -0,0 +1,372 @@
...
+                break
+
+            if self.stream is None:
+                with hubblestack.utils.asynchronous.current_ioloop(self.io_loop):
+                    self.stream = IOStream(
+                        socket.socket(sock_type, socket.SOCK_STREAM)
+                    )
+            try:
+                log.trace('IPCClient: Connecting to socket: %s', self.socket_path)
+                yield self.stream.connect(sock_addr)
```
Comment (goravsingal) [inline hubblestack/transport/ipc.py:129]: I hope we are copying code which we are using.
It seems, this code is related to salt/minion which we don't use.

File: hubblestack/transport/ipc.py [hunk 5]
```diff
@@ -0,0 +1,372 @@
...
+import errno
+import logging
+import socket
+import time
+
+# Import 3rd-party libs
+import msgpack
+
+# Import Tornado libs
+import tornado
```
Comment (goravsingal) [inline hubblestack/transport/ipc.py:17]: I think we need to add these kind of dependencies in our requirements.txt file as well.
Else, we will face lot of such issues in the end when we will remove salt completely.

File: hubblestack/utils/decorators/jinja.py [hunk 6]
```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
```
Comment (goravsingal) [inline hubblestack/utils/decorators/jinja.py]: Need to remove usage of Jinja filter. We don't need this.

File: hubblestack/utils/event.py [hunk 7]
```diff
@@ -0,0 +1,851 @@
+# -*- coding: utf-8 -*-
+'''
+Manage events
+
+Events are all fired off via a zeromq 'pub' socket, and listened to with local
```
Comment (goravsingal) [inline hubblestack/utils/event.py:5]: I believe, we dont need this file either. We don't use events (zeromq)

File: hubblestack/utils/hashutils.py [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+# Import python libs
+import hashlib
+
+import hubblestack.utils.files
+
+from hubblestack.utils.decorators.jinja import jinja_filter
+
+
+@jinja_filter('file_hashsum')
```
Comment (goravsingal) [inline hubblestack/utils/hashutils.py]: we can remove this annotation

File: hubblestack/utils/raetevent.py [hunk 9]
```diff
@@ -0,0 +1,281 @@
+# -*- coding: utf-8 -*-
+'''
+Manage events
```
Comment (goravsingal) [inline hubblestack/utils/raetevent.py:3]: Please check if this file is relevant. We don't use events

**Review by goravsingal — CHANGES_REQUESTED** [review]

File: hubblestack/config/__init__.py (hunk 1 above)
Comment (devagupt) [reply hubblestack/config/__init__.py]: Removed config as it will be ported by Paul.

File: hubblestack/payload.py (hunk 2 above)
Comment (devagupt) [reply hubblestack/payload.py]: Removed payload as it will be ported by Paul.

File: hubblestack/payload.py (hunk 3 above)
Comment (devagupt) [reply hubblestack/payload.py]: Not ported.

File: hubblestack/transport/ipc.py (hunk 5 above)
Comment (devagupt) [reply hubblestack/transport/ipc.py:17]: Added tornado in dependency

File: hubblestack/utils/decorators/jinja.py (hunk 6 above)
Comment (devagupt) [reply hubblestack/utils/decorators/jinja.py]: Removed.

File: hubblestack/utils/raetevent.py (hunk 9 above)
Comment (devagupt) [reply hubblestack/utils/raetevent.py:3]: This file is used by event utility which is needed in roots fileserver.

File: hubblestack/utils/hashutils.py (hunk 8 above)
Comment (devagupt) [reply hubblestack/utils/hashutils.py]: Removed

File: hubblestack/utils/event.py (hunk 7 above)
Comment (devagupt) [reply hubblestack/utils/event.py:5]: Events is needed in roots fileserver

**Pull request merged**